yt_dlp/extractor/twitch.py

```diff
@@ -198,7 +198,9 @@ def _extract_twitch_m3u8_formats(self, video_id, token, signature):
                 'allow_audio_only': 'true',
                 'allow_spectre': 'true',
                 'p': random.randint(1000000, 10000000),
+                'platform': 'web',
                 'player': 'twitchweb',
+                'supported_codecs': 'av1,h265,h264',
```

Comment:
From some testing, it seems that yt-dlp (ffmpeg) cannot successfully download an av1 twitch livestream. If we add this format, it should be deprioritized

Reply:
What was the issue? I've tested it in mpv with latest ffmpeg. Both hevc and av1 worked fine. What does "cannot successfully download" mean exactly?

Reply:
Downloading with ffmpeg 6.1.1 resulted in a file that plays back w/o video. From the log:
```
[mpegts @ 000002209f939400] Stream 0, codec av1, is muxed as a private data stream and may not be recognized upon reading.
```
which leads me to:
https://old.reddit.com/r/ffmpeg/comments/13ocph8/hls_segmentation_av1/

Reply:
In my testing, the resulting files were missing the video stream. I did not have time to investigate this further. With only very few channels having access to the beta, it is also quite difficult to test.

Here is a log I recorded: [av1_log_twitch.txt](https://github.com/yt-dlp/yt-dlp/files/14317586/av1_log_twitch.txt)
Given this line:
> [mpegts @ 000002209f939400] Stream 0, codec av1, is muxed as a private data stream and may not be recognized upon reading.

it seems most likely to me that AV1 inside MPEG-TS is not officially supported, so ffmpeg adds the av1 video as a private data stream. This then gets lost in the final remuxing step.

Reply:
> If there is an issue, I would prefer fixing it on ffmpeg side.

Even if it is fixed in ffmpeg master tomorrow, we can't prioritize a format that doesn't work with the vast majority of ffmpeg installations over the stable formats our users expect.

Reply:
This is log from mpegts muxer. Maybe its should be saved as mp4 or mkv in case of AV1. And I think this is general yt-dlp for AV1 streams that currently mpegts support is not fully completed.

Reply:
> Even if it is fixed in ffmpeg master tomorrow, we can't prioritize a format that doesn't work with the vast majority of ffmpeg installations over the stable formats our users expect.

Of course, I was just thinking there is some bug in ffmpeg with those streams that should be fixed, but it is like missing feature, which is not likely to be finalized anytime soon.

Reply:
we could try something like this, but I can't find any streams to test with rn (and I don't know if `startswith('av1')` will be effective)
```diff
diff --git a/yt_dlp/extractor/twitch.py b/yt_dlp/extractor/twitch.py
index 6dc0993af..c14bb4395 100644
--- a/yt_dlp/extractor/twitch.py
+++ b/yt_dlp/extractor/twitch.py
@@ -192,7 +192,7 @@ def _get_thumbnails(self, thumbnail):
 
     def _extract_twitch_m3u8_formats(self, video_id, token, signature):
         """Subclasses must define _M3U8_PATH"""
-        return self._extract_m3u8_formats(
+        formats = self._extract_m3u8_formats(
             f'{self._USHER_BASE}/{self._M3U8_PATH}/{video_id}.m3u8', video_id, 'mp4', query={
                 'allow_source': 'true',
                 'allow_audio_only': 'true',
@@ -203,6 +203,11 @@ def _extract_twitch_m3u8_formats(self, video_id, token, signature):
                 'sig': signature,
                 'token': token,
             })
+        for fmt in formats:
+            if fmt.get('vcodec') and fmt['vcodec'].startswith('av1'):
+                fmt['downloader_options'] = {'ffmpeg_args': ['-f', 'mp4']}
+
+        return formats
```

Reply:
nvm, `ffmpeg_args` are added as input args rather than output args so that won't work

Reply:
> nvm, `ffmpeg_args` are added as input args rather than output args so that won't work

You can add a new `downloader_options`

Reply:
could we add a dict that works similar to how the CLI does? e.g.
```py
fmt['downloader_options'] = {
    'ffmpeg': {
        'i1': [],
        'i2': [],
        'o': ['-f', 'mp4'],
    },
}
```

Reply:
`i1`,`i2` args will be unusable since extractor don't know what number is what format. So easier just to have a `ffmpeg_args_output` imo